src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.ValidateToken.Internal.cs

```diff
@@ -36,19 +36,32 @@ internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
         }
 
         internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
-            Saml2SecurityToken samlToken,
+            SecurityToken securityToken,
             ValidationParameters validationParameters,
             CallContext callContext,
             CancellationToken cancellationToken)
         {
-            if (samlToken is null)
+            if (securityToken is null)
             {
                 StackFrames.TokenNull ??= new StackFrame(true);
                 return ValidationError.NullParameter(
-                    nameof(samlToken),
+                    nameof(securityToken),
                     StackFrames.TokenNull);
             }
 
+            if (securityToken is not Saml2SecurityToken samlToken)
```

Comment:
`securityToken is not SamlSecurityToken samlToken` -> `securityToken is not SamlSecurityToken`

Reply:
Replied above.